digicampipe/io/event_stream.py

```diff
@@ -1,25 +1,112 @@
+    for file in file_list:
         if not mc:
+            data_stream = zfits.zfits_event_source(
+                url=file,
+                expert_mode=expert_mode,
+                camera_geometry=camera_geometry,
+                max_events=max_events,
+                camera=camera
+            )
+        else:
+            data_stream = hdf5.digicamtoy_event_source(
+                url=file,
+                camera_geometry=camera_geometry,
+                camera=camera,
+                max_events=max_events
+            )
+        for event in data_stream:
+            yield event
+
 
-            data_stream = zfits.zfits_event_source(url=file,
-                                                   expert_mode=expert_mode,
-                                                   camera_geometry=camera_geometry,
-                                                   max_events=max_events,
-                                                   camera=camera)
+def get_slow_data_info(file_list):
+    if len(file_list) == 0:
+        print("ERROR: no slow data file given")
+        return
+    data_structs={}
+    # get basic information from slow data
+    # (class, min and max timestamp, data location)
+    for file in file_list:
+        data_struct = {}
+        hdulist = fits.open(file)
+        data_struct['hdu'] = hdulist[1]
+        ts = data_struct['hdu'].data['timestamp']
+        good = ts != 0
+        events = np.arange(len(ts))
+        data_struct['timestamps'] = ts[good]
+        data_struct['events'] = events[good]
+        data_struct['ts_min'] = min(data_struct['timestamps'])
+        data_struct['ts_max'] = max(data_struct['timestamps'])
+        filename = os.path.basename(file)
+        m = re.match('(?:slow_)?(?P<class>[\w]+?)_[\d\_]+\.fits',
+                     filename)
+        class_name = m.group("class")
+        if class_name not in data_structs.keys():
+            data_structs[class_name]=[data_struct]
         else:
+            data_structs[class_name].append(data_struct)
+    return data_structs
 
-            data_stream = hdf5.digicamtoy_event_source(url=file,
-                                                       camera_geometry=camera_geometry,
-                                                       camera=camera,
-                                                       max_events=max_events)
 
-        for event in data_stream:
+def get_slow_event(info_struct_list, data_ts):
+    ts_min_all = [info_struct['ts_min'] for info_struct in info_struct_list]
+    ts_max_all = [info_struct['ts_max'] for info_struct in info_struct_list]
+    files = [info_struct['hdu']._file.name for info_struct in info_struct_list]
+    ts_min_all = np.array(ts_min_all)
+    ts_max_all = np.array(ts_max_all)
+    files = np.array(files)
+    indexes_slow_file = np.logical_and(
+        ts_min_all < data_ts,
+        ts_max_all > data_ts
+    )
+    if np.sum(indexes_slow_file) == 0:
+        # print('ERROR: no slow file found !')
```

Comment:
The print/warning should be restored I think. Otherwise there is no way to see want went wrong?

Reply:
the warning is issued in add_slow_data(), that why I disabled it here.